modules/consensus/consensusset.go

```diff
@@ -211,6 +211,9 @@ func (cs *ConsensusSet) Close() error {
 
 // CurrentBlock returns the latest block in the heaviest known blockchain.
 func (cs *ConsensusSet) CurrentBlock() (block types.Block) {
+	cs.mu.RLock()
+	defer cs.mu.RUnlock()
```

Comment:
That ends up being difficult to answer. It is protecting the `cs.db`, but if we add locking to the other exported functions, we'll get deadlocks. Now that we have a `ThreadGroup`, a lot of this stuff can be repaired more easily, but going in to depth on everything is going to be a pull request of its own.

=/